src/Caliburn.Micro.Avalonia/Platforms/netcore-avalonia/WindowManager.cs

```diff
@@ -0,0 +1,303 @@
+                //    popup.AllowsTransparency = true;
+                //}
+            }
+            else
+            {
+                //popup.AllowsTransparency = true;
+                popup.PlacementMode = PlacementMode.Pointer;
+            }
+
+            return popup;
+        }
+
+        /// <summary>
+        /// Creates a window.
+        /// </summary>
+        /// <param name="rootModel">The view model.</param>
+        /// <param name="isDialog">Whethor or not the window is being shown as a dialog.</param>
+        /// <param name="context">The view context.</param>
+        /// <param name="settings">The optional popup settings.</param>
+        /// <returns>The window.</returns>
+        public virtual async Task<Window> CreateWindowAsync(object rootModel, bool isDialog, object context, IDictionary<string, object> settings)
+        {
+            var view = EnsureWindow(rootModel, ViewLocator.LocateForModel(rootModel, null, context), isDialog);
+            ViewModelBinder.Bind(rootModel, view, context);
+
+            var haveDisplayName = rootModel as IHaveDisplayName;
+            if (string.IsNullOrEmpty(view.Title) && haveDisplayName != null && !ConventionManager.HasBinding(view, Window.TitleProperty))
+            {
+                var binding = new Binding("DisplayName") { Mode = BindingMode.TwoWay };
+                view.Bind(Window.TitleProperty, binding);
+            }
+
+            ApplySettings(view, settings);
+
+            var conductor = new WindowConductor(rootModel, view);
+
+            await conductor.InitialiseAsync();
+
+            return view;
+        }
+
+        /// <summary>
+        /// Makes sure the view is a window is is wrapped by one.
+        /// </summary>
+        /// <param name="model">The view model.</param>
+        /// <param name="view">The view.</param>
+        /// <param name="isDialog">Whethor or not the window is being shown as a dialog.</param>
+        /// <returns>The window.</returns>
+        protected virtual Window EnsureWindow(object model, object view, bool isDialog)
+        {
+
+            if (view is Window window)
+            {
+                var owner = InferOwnerOf(window);
+                if (owner != null && isDialog)
+                {
+                    //TODO: (Avalonia) Set window owner
+                    //window.Owner = owner;
+                }
+            }
+            else
+            {
+                window = new Window
+                {
+                    Content = view,
+                    SizeToContent = SizeToContent.WidthAndHeight
+                };
+
+                window.SetValue(View.IsGeneratedProperty, true);
+
+                var owner = InferOwnerOf(window);
+                if (owner != null)
+                {
+                    window.WindowStartupLocation = WindowStartupLocation.CenterOwner;
+                    //TODO: (Avalonia) Set window owner
+                    //window.Owner = owner;
```

Comment:
do we need to set window owner here?

Reply:
judging from the methods name, it should. 
why  has nobody answered your review-questions for such a long time?

Reply:
I think the question is because this flow is not valid for Avalonia.
You pass the owner as parameter to ShowDialog (mandatory) or Show (optional). So it is really a business/architectural decision if we divert from traditional Caliburn way of doing it at the cost of introducing inconsistencies between platforms.

Reply:
Also beware that work has continued here: https://github.com/Caliburn-Micro/Caliburn.Micro/tree/556-compatibility-inquiry-avaloniaui
It can use a rebase btw cause it is not compatible anymore with current master, there are conflicts due to merge of MAUI support.